src/settings/window_geometry.rs

```diff
@@ -1,16 +1,64 @@
-#[derive(Debug, Clone)]
+use crate::renderer::Renderer;
+use crate::settings::SETTINGS;
+use crate::window::WindowSettings;
+use glutin::dpi::PhysicalSize;
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
 pub struct WindowGeometry {
     pub width: u64,
     pub height: u64,
 }
 
+#[cfg(unix)]
+const SETTINGS_PATH_EXTENSION: &str = ".local/share/nvim/neovide-settings.json";
+#[cfg(windows)]
+const SETTINGS_PATH_EXTENSION: &str = "AppData/Local/nvim-data/neovide-settings.json";
+
+fn neovim_std_datapath() -> PathBuf {
+    let mut settings_path = dirs::home_dir().unwrap();
+    settings_path.push(SETTINGS_PATH_EXTENSION);
+    settings_path
+}
+
+pub fn try_to_load_last_window_size() -> Result<WindowGeometry, String> {
+    let settings_path = neovim_std_datapath();
+    let serialized_size = std::fs::read_to_string(&settings_path).map_err(|e| e.to_string())?;
+
+    let saved_window_size: WindowGeometry =
+        serde_json::from_str(&serialized_size).map_err(|e| e.to_string())?;
+    log::debug!("Loaded Window Size: {:?}", saved_window_size);
+    Ok(saved_window_size)
+}
 pub const DEFAULT_WINDOW_GEOMETRY: WindowGeometry = WindowGeometry {
     width: 100,
     height: 50,
 };
 
+pub fn maybe_save_window_size(window_size: PhysicalSize<u32>, renderer: &Renderer) {
+    let saved_window_size = if SETTINGS.get::<WindowSettings>().remember_window_size {
+        WindowGeometry {
+            width: (window_size.width as f32 / renderer.font_width as f32) as u64,
+            height: (window_size.height as f32 / renderer.font_height as f32) as u64,
+        }
+    } else {
+        WindowGeometry {
+            width: DEFAULT_WINDOW_GEOMETRY.width as u64,
+            height: DEFAULT_WINDOW_GEOMETRY.height as u64,
+        }
+    };
+
+    let settings_path = neovim_std_datapath();
+    let se = serde_json::to_string(&saved_window_size).unwrap();
+    //log::debug!("Saved Window Size: {}", se);
```

Comment:
```suggestion
    log::debug!("Saved Window Size: {}", se);
```

Maybe remove both, or leave both log:debug? no need to leave it commented.

Reply:
Kk